### ISSUE TITLE
libmpv freeze when constantly creating and destroying multiple videos.

### ISSUE THREAD

**Issue by maiw** [mpv-player/mpv#4891]
### mpv version and platform
mpv-1.dll   0.27.0 
Windows 10

### Reproduction steps
Repeatedly create/destroy multiple videos using libmpv in C#.

### Expected behavior
The videos get created and then later destroyed over and over.

### Actual behavior
Behaves as expected, but after some variable amount of time, it freezes.

When it's frozen, clicking the pause button in visual studio shows that the next statement to execute once it finishes what it's doing is one of the mpv_terminate_destroy calls (I have 8 of them called in a row, one for each video).

There also seems to be some sort of memory leak in the sample project that you can notice if it lasts long enough without freezing. This leak did not exist in 0.26.0

### Log file
Log files from the last 8 videos when the freeze happened.
[log 1309.txt](https://github.com/mpv-player/mpv/files/1311101/log.1309.txt)
[log 1310.txt](https://github.com/mpv-player/mpv/files/1311102/log.1310.txt)
[log 1311.txt](https://github.com/mpv-player/mpv/files/1311100/log.1311.txt)
[log 1312.txt](https://github.com/mpv-player/mpv/files/1311099/log.1312.txt)
[log 1313.txt](https://github.com/mpv-player/mpv/files/1311103/log.1313.txt)
[log 1314.txt](https://github.com/mpv-player/mpv/files/1311104/log.1314.txt)
[log 1315.txt](https://github.com/mpv-player/mpv/files/1311105/log.1315.txt)
[log 1316.txt](https://github.com/mpv-player/mpv/files/1311098/log.1316.txt)

### Sample files
[mpv-dll-example.zip](https://github.com/mpv-player/mpv/files/1311139/mpv-dll-example.zip)

**Comment by wm4** [comment]
Could be an issue in our pthreads wrapper. Can you try with a libmpv compiled with libwinpthreads instead of the native wrapper?

**Comment by maiw** [comment]
Thank you very much for the quick response.

My attempts at compiling libmpv have failed. Even at best, it only created the dll with 1/3'rd the normal filesize that it should be when compared to the normal one. Attempting to load it in the sample project fails.

Do you know if there is any libmpv already compiled with that option that I could use?

**Comment by fitipe** [comment]
Hi there,
I'm probably facing a similar problem, running mpv on a Raspberry pi 3 B.

I've been using a software called puredata to call mpv and play videos randomly. What it does is to periodically send a message ( mpv  /my/file/directory/file.mov  ) to an object called shell that throws it into the terminal.

I also set --speed,  --start, --length with randomly generated values. 

It plays beautifully for 5 to 10 minutes and then it freezes on a frame. Puredata keeps running behind and I can hear the audio of the videos, still selected ramdomly, but all I can see is the same frozen image. Another problem ( always present ) is that while any videos are played with this mechanism, I have no interactive controls. So, when it freezes, I have no option left than pulling the raspberry's energy plug and turn it on again.

Trying to figure out possible causes, I accessed the raspberry through a vnc server and launched the program using my notebook connected with a network cable, leaving the hdmi to the television. So I could watch the videos running on the tv and visualize puredata's patch at the same time. So, on the tv there were the videos playing one after another and, on the notebook's screen, there was only the patch running.

It froze again but, unexpectedly, mpv was forced to switch it's main VO option and begun to run it into the vnc-server's screen, on my notebook. And never froze at this one. The audio kept been played on the television! Also, on the vnc-server's screen, I have all the interactive controls back.

I found out that, when it freezes, mpv is no longer capable of setting te video output to "rpi". Instead, it tries to set it to a bunch of options and the only possible one is "x11" - which runs on the vnc server, opened on my notebook.

Do some one have any ideias? 
I have no understanding on coding, but I've been digging around a lot trying to find it out ...

Thanks a lot!
